Source/ACE.Database/CharacterDatabase.cs

```diff
@@ -151,12 +164,14 @@ public bool IsNameAvailable(string name)
             return (charsWithName == 0);
         }
 
-        public Task UpdateCharacter(Character character)
+        public async void UpdateCharacter(Character character)
```

Comment:
Changed this function to async, to allow the occurrence of events independent of the main program flow. The idea was mainly derived from the CreateCharacter function below.